source/index.rst

```diff
@@ -6,12 +6,55 @@
 Welcome to ds-fastapi-middleware's documentation!
 =================================================
 
+ds-fastapi-middleware is a Python project that provides middleware components for FastAPI applications. This middleware is designed to enhance the functionality and performance of your FastAPI application by providing common features such as context, usage, audit and timeout.
+
+High-Level Overview
+-------------------
+FastAPI is a modern, fast (high-performance), web framework for building APIs with Python 3.7+ based on standard Python type hints. Middleware in FastAPI allows you to process requests before they reach your endpoint logic and responses before they are sent to the client. Middleware can be used to implement cross-cutting concerns like authentication, logging, error handling, and more.
+
+Key Features
+------------
+- **Audit Logging**: Automatically log incoming requests.
+- **Context Management**: Manage request context data.
+- **Timeout Handling**: Set a timeout for incoming requests and handle timeouts gracefully.
+- **Service Usage**: Track and report usage statistics for your API.
+
+Getting Started
+---------------
+To get started with ds-fastapi-middleware, you need to have Python and FastAPI installed. You can install FastAPI and other dependencies using pip:
+
+.. code-block:: bash
+
+    pip install fastapi
+    pip install uvicorn  # ASGI server for FastAPI
+
+Then, you can create a simple FastAPI application and add middleware to it:
+
+.. code-block:: python
+
+    from fastapi import FastAPI
+    from ds_fastapi_middleware import ExampleMiddleware
+
+    app = FastAPI()
+
+    app.add_middleware(ExampleMiddleware)
```

Comment:
This is nitpicking, but maybe it would be better to use an actual middleware rather than `ExampleMiddleware` that way it will be possible to just copy paste and run the example.